.github/workflows/publish-test.yml

```diff
@@ -1,105 +1,93 @@
-# This workflow will upload a Python Package to Release asset
-# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
-
 name: Publish to Test PyPI
 
 on:
   push:
     branches:
       - main
 
-# Needed to create release and upload assets
 permissions:
   contents: write
 
-
 jobs:
   setup-version:
     runs-on: ubuntu-latest
     steps:
-    - name: Generate version number
-      run: |
-        VERSION_HASH=$(date +"%Y%m%d%H%M%S")
-        echo "Generated version hash: $VERSION_HASH"
-        echo $VERSION_HASH > version.txt
+      - name: Generate version number
+        run: |
+          VERSION_HASH=$(date +"%Y%m%d%H%M%S")
+          echo "Generated version hash: $VERSION_HASH"
+          echo $VERSION_HASH > version.txt
 
-    - name: Upload version number as artifact
-      uses: actions/upload-artifact@v4
-      with:
-        name: version
-        path: version.txt
+      - name: Upload version number as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: version
+          path: version.txt
 
   wheel:
     name: Build Wheel
-    runs-on: ${{ matrix.os }}
-    permissions: write-all
-
+    runs-on: ubuntu-20.04
+    container:
+      image: nvidia/cuda:12.1.1-cudnn8-devel-ubuntu20.04
+      options: --gpus all
```

Comment:
We don't have gpus in github actions runner, this option might better be removed